File: server/src/main/scala/kpn/api/common/subset/SubsetFactDetailsPage.scala
package kpn.api.common.subset

import kpn.api.common.Fact

case class SubsetFactDetailsPage(subsetInfo: SubsetInfo, fact: Fact, networks: Seq[NetworkFactRefs]) {
  def refCount: Int = networks.map { n => n.factRefs.size }.sum
}

File: backend/api/src/main/java/kpn/api/common/subset/SubsetFactDetailsPage.java
package kpn.api.common.subset;

import kpn.api.common.Fact;
import kpn.api.common.subset.NetworkFactRefs;
import kpn.api.common.subset.SubsetInfo;

import com.google.common.collect.ImmutableList;

public record SubsetFactDetailsPage(
  SubsetInfo subsetInfo,
  Fact fact,
  ImmutableList<NetworkFactRefs> networks
) {}

/* TODO migrate

  def refCount: Int = networks.map { n => n.factRefs.size }.sum

*/
